### ISSUE TITLE
Accessibilty Testing - Core - Table

### ISSUE THREAD

**Issue by george861** [SAP/fundamental-ngx#6178]
Accessibility: Table -Core Component (There are couple of accessibility issues found while testing, so i have attached error document for you reference.)
Screenshot:
[Table  Core_Ariba Controls.docx](https://github.com/SAP/fundamental-ngx/files/6918739/Table.Core_Ariba.Controls.docx)

Table - Core

1.1	Message Details

Raise Ticket	
Category	
Preprocessed By	
Description	

1.2 Application & Test Environment Details

Application Name:	ARIBA Controls- Fundamental Library for Angular
URL:	https://fundamental-ngx.netlify.app/#/core/panel

Technology:	Angular 11
Web browser:	Google Chrome Version 92.0.4515.107 (Official Build) (64-bit)
Operating System:	Microsoft Windows 10 x64
Screen Reader:	JAWS 2021.2105.53 ILM

 

Issue 1

Priority:	Medium
Checkpoint:	ACC-261

(261.3)

•	Search ‘Table’ from the left panel
Observed behavior:
The links such as ‘row 1’ within the column ‘column 1’ within the tables doesn’t suffice the minimal contrast ratio of 7:1 in high contrast black theme

Expected Behavior:    
The links such as ‘row 1’ within the column ‘column 1’ within the tables should suffice the minimal contrast ratio of 7:1 in high contrast black theme

Screenshot 1  
![image](https://user-images.githubusercontent.com/88019003/127781928-ea74d7e6-5f2e-4290-b262-c6efce75c731.png)

Case 2
![image](https://user-images.githubusercontent.com/88019003/127781930-8fa6cadf-f958-446c-b875-523b87d967e5.png)

 

Issue 2

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Table’ from the left panel

Observed Behavior:
Case 1
On accessing any table, no information is conveyed about the table title, the type of the table such as ‘multi – selectable table’ and the total number of columns and rows it contains. 
Case 2
On accessing the links such as ‘user.name@email.com’ within the column ‘Column header’ in the ‘Default table’, the associated column header information and the row count are not conveyed.

Expected Behavior:
Case 1
On accessing any table, the table title, the type of the table such as ‘multi – selectable table’ and the total number of columns and rows it contains should be announced. 
Case 2
On accessing the links such as ‘user.name@email.com’ within the column ‘Column header’, the associated column header information should be conveyed along with the row count.

NOTE: the issues persists in all similar tables

Screenshot:
Case 1
 
![image](https://user-images.githubusercontent.com/88019003/127781950-e6d76581-2418-4fd5-9dde-63c183c0cd92.png)

Case 2
 
![image](https://user-images.githubusercontent.com/88019003/127781953-d7592f6b-f8c5-42ed-8bbf-ab48c08e5f8c.png)

Case 3
 
![image](https://user-images.githubusercontent.com/88019003/127781958-84e0d7d4-0277-48c0-8d04-59b0eb520378.png)

Issue 3
Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Table’ from the left panel

Observed Behavior: 
Case 1
On accessing the any table, no information is conveyed about table title and its type and the total columns and rows available within the table. 
Case 2
On accessing any controls such as link within the table, the corresponding column header is not associated. 
Case 3
On accessing any data cell within the table, the associated column and row information not conveyed.

Expected Behavior: 
Case 1
On accessing any table within the application, the information should be conveyed about table header and its type such as ‘Multi selectable table’ and the total columns and rows available within the table.
Case 2
On accessing any link such as ‘usr.name@email.com’ within the table, the column header should be associated. 
Case 3
On accessing the data cells within the tables, the associated row and column information should be conveyed

 Screenshot:
![image](https://user-images.githubusercontent.com/88019003/127781964-4f64c226-fc10-41dc-9fc3-8cf6327a57c8.png)

Issue 4

Priority:	Medium
Checkpoint:	ACC-255

(255.1)

•	Search ‘Table’ from the left panel
Observed behavior:
No tooltip or text alternative provided for the checkboxes within the table
Expected Behavior:    
Appropriate text alternative should be provided for the checkboxes within the table

Screenshot 1 

![image](https://user-images.githubusercontent.com/88019003/127781980-c9815128-f5e5-4649-a836-a1b1180a343d.png)

 Case 2
 
![image](https://user-images.githubusercontent.com/88019003/127781983-740ec0d5-4b58-45f0-b657-44579d9a2844.png)

NOTE: the issues persists in all similar tables

Issue 5

Priority:	High
Checkpoint:	ACC-264, ACC-270

(264.1)(270.13)

•	Search ‘Table’ from the left panel
Observed behavior:
On accessing the ‘New item’ dialog, the initial focus is provided on the ‘cancel’ button which is not the first interactive element within the dialog.
The dialog title ‘New Item’ is not announced 
Expected Behavior:   
On accessing the ‘New item’ dialog, the initial focus is provided on the first interactive element within the dialog and the dialog should also be announced.

Screenshot 1 

 
![image](https://user-images.githubusercontent.com/88019003/127781989-3c24f549-a876-4c2b-9f21-aa45a7551f67.png)

Issue 6

Priority:	High
Checkpoint:	ACC-264, ACC-270

(264.1) (270.1)

•	Search ‘Table’ from the left panel
Observed behavior:
On accessing the column headers such as ‘Column 1’ within the table ‘Column Sorting and Filtering’, the sorted information is not conveyed.
No information is conveyed about the column headers as it can be activated via keyboard. 
The buttons within column headers such as ‘Sort Ascending’, ‘Sort Descending’ and ‘Filter’ are not accessible via keyboard. 

Expected Behavior:    
On accessing the column headers such as ‘Column 1’ within the table ‘Column Sorting and Filtering’, the sorted information should be conveyed.
The information should be conveyed to the user about the column headers as it can be activated via keyboard and user can sort and filter the table.
The buttons within column headers such as ‘Sort Ascending’, ‘Sort Descending’ and ‘Filter’ should be accessible via keyboard.
Screenshot 1 
 
![image](https://user-images.githubusercontent.com/88019003/127781999-a8d197f5-33b3-44d4-9f52-1b4a1f54cd90.png)

Screenshot 2
 
![image](https://user-images.githubusercontent.com/88019003/127782004-ffb882c2-fc15-495d-8811-742212e08685.png)

Issue 7

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Table’ from the left panel
Observed behavior:
No information is conveyed about the status of the table row as ‘valid’,’warning’, ‘information’, and ‘error’. 
Speech:
column 1 row 4
check box  not checked  
To check press Spacebar.
Column 1
user.name@email.com Link 
Expected Behavior:    
the information should be conveyed about the status of the table row as ‘valid’,’warning’, ‘information’, and ‘error’ on accessing to the screen reader user.

Screenshot 1 
 
![image](https://user-images.githubusercontent.com/88019003/127782009-276104b3-727e-4ddd-8e5d-08362f77761e.png)

Issue 8

Priority:	High
Checkpoint:	ACC-270

(270.3)

•	Search ‘Table’ from the left panel

Observed Behavior: 
The navigation within the table ‘Focusable’ is only possible via tabbing which is not an efficient way to navigate within table as if 40 columns then the user needs to navigate through all 40 columns every time to reach or access the table row.
Expected Behavior:    
The table data cells should be accessible via arrow keys as when the table comprises a greater number of columns and rows. Thus, for the user to reach the row such as 20th row within the table needs to press tab 40 times or more to reach which is not an efficient way to access the table contents.  
The navigation within the table ‘Focusable’ should be possible via arrow keys which will help the user to reach or to navigate within the table in a more efficient way.
Screenshot 1 
![image](https://user-images.githubusercontent.com/88019003/127782015-8340b55d-43fe-462d-b655-cbbb57d81514.png)

 
Issue 9

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Table’ from the left panel
Observed behavior:
On accessing the table ‘Table with Angular CDK’, an additional information ‘Grid’ is announced.
On accessing the link ‘Row 1’, the column header and the row count information are not announced.
Speech:
Grid  
Row 1 Link 
Expected Behavior:    
On accessing the any table, only meaningful and relevant information should be conveyed.
On accessing the link ‘Row 1’ within the column ‘Column 1’,the column header and the row count information should also be announced.

Screenshot 1 
 
![image](https://user-images.githubusercontent.com/88019003/127782020-e4bd864f-6091-49a5-bcc1-4852e0b97eaf.png)

Issue 10

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Table’ from the left panel
Observed behavior:
On accessing the checkbox within the column header, the column header ‘Fruit Description’ and ‘Price’ is not associated
On accessing the checkboxes within the table row, the associated row information is not conveyed.

Expected Behavior:    
On accessing the checkbox within the column header, the column header ‘Fruit Description’ and ‘Price’ should be associated
On accessing the checkboxes within the table row, the associated row information should be conveyed.
Screenshot 1 
 
![image](https://user-images.githubusercontent.com/88019003/127782027-435ef25c-7bc4-4ad6-a874-35411047005f.png)

Case 2
 
![image](https://user-images.githubusercontent.com/88019003/127782029-b7596e64-5e52-4ff1-974b-55585ec6b74c.png)

Issue 11

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Table’ from the left panel
Observed behavior:
In reading mode, on accessing the image within the column ‘column header’ in the table ‘Table with pagination’, no information is conveyed.
Expected Behavior:    
In reading mode, on accessing the image within the column ‘column header’ in the table ‘Table with pagination’, the related information should be conveyed.
Screenshot 1 

 
![image](https://user-images.githubusercontent.com/88019003/127782032-f88dbce8-e6a9-4c36-b42d-05be8aa0c8c0.png)

Issue 12

Priority:	Medium
Checkpoint:	ACC-255

(255.1)

•	Search ‘Table’ from the left panel
Observed behavior:
No text alternative or tooltip provided for the buttons within the table ‘Row with navigation button’
Expected Behavior:    
Appropriate text alternative should be provided for the buttons within the table ‘Row with navigation button’
Screenshot 1 
![image](https://user-images.githubusercontent.com/88019003/127782034-9cabda71-774e-491c-a3ad-58ae00a7a056.png)

Issue 13

Priority:	Medium
Checkpoint:	ACC-264

(264.1)

•	Search ‘Table’ from the left panel
Observed behavior:
On accessing the buttons within the table ‘Row with navigation button’, the column count information conveyed is incorrect.
Expected Behavior:    
On accessing the buttons within the table ‘Row with navigation button’, the column count information should be correctly conveyed.

Screenshot 1 
![image](https://user-images.githubusercontent.com/88019003/127782038-7ff06079-2cf6-42c9-8425-b748358169d4.png)

**Comment by Karthickc1994** [comment]
Hello Colleague,

Retest has been taken below are the test results.


Issue1: Fixed
Issue2: Fixed
Issue3: Fixed
Issue4: reproducible 
•	Search ‘Table’ from the left panel
Observed behavior:
No tooltip or text alternative provided for the checkboxes within the table
Expected Behavior:    
Appropriate text alternative should be provided for the checkboxes within the table
![image](https://user-images.githubusercontent.com/105037808/172532757-304b08cd-b8d7-494c-96a2-0d62cf2031b9.png)

 

Case2:
 
![image](https://user-images.githubusercontent.com/105037808/172532729-a295bc54-027c-4af6-8d76-0b771f6d764a.png)


Issue5: Fixed
Issue6:

Observed behaviour:
Upon changing the selection in shorting the table column header additional information is read out.
Expected behaviour: 
Upon changing the selection in shorting
Screen shot: 
![image](https://user-images.githubusercontent.com/105037808/172532701-e9fa50df-4113-4f95-b0ef-6c9828ff1257.png)
 
Issue7: Fixed
Issue8: Fixed
Issue9: Fixed
Issue10:Fixed
Issue11:Fixed
Issue12: Fixed
Issue13: Fixed


Retest environment:
Preview URL: https://fundamental-ngx-gh--pr8184-ds-6178-core-table-a-1lvvdjey.web.app/
Browser: Chrome Version 102.0.5005.63 (Official Build) (64-bit)
Screen Reader: JAWS 2022.2204.20ILM

Regards,
Karthikeyan.c

**Comment by Karthickc1994** [comment]
Hello Colleague,
Retest has been taken below are the test results.

Issue1: Fixed
Issue2: Fixed
Issue3: Fixed
Issue4: Fixed
Issue5: Fixed
Issue6:
Observed behaviour:
Upon changing the selection in shorting the table column header additional information is read out.
Expected behaviour:
Upon changing the selection in shorting
Screen shot:
![image](https://user-images.githubusercontent.com/105037808/174255962-b4d3b055-cfd5-4eef-a969-2d11590525c1.png)

Issue7: Fixed
Issue8: Fixed
Issue9: Fixed
Issue10:Fixed
Issue11:Fixed
Issue12: Fixed
Issue13: Fixed

Retest environment:
preview URL: •	https://deploy-preview-8184--fundamental-ngx.netlify.app/#/core/table
Browser: Chrome Version 102.0.5005.63 (Official Build) (64-bit)
Screen Reader: JAWS 2022.2204.20ILM

Regards,
Karthikeyan.c

**Comment by Karthickc1994** [comment]
Hello Colleague,
Retest has been taken below are the test results.

Issue1: Fixed
Issue2: Fixed
Issue3: Fixed
Issue4: Fixed
Issue5: Fixed
Issue6: Fixed
Issue7: Fixed
Issue8: Fixed
Issue9: Fixed
Issue10:Fixed
Issue11:Fixed
Issue12: Fixed
Issue13: Fixed

Retest environment:
preview URL: • https://deploy-preview-8184--fundamental-ngx.netlify.app/#/core/table
Browser: Chrome Version 102.0.5005.63 (Official Build) (64-bit)
Screen Reader: JAWS 2022.2204.20ILM

Regards,
Karthikeyan.c
